pkg/reconciler/buildrun/buildrun.go

```diff
@@ -323,90 +358,74 @@ func (r *ReconcileBuildRun) VerifyRequestName(ctx context.Context, request recon
+			return nil, resources.HandleError("failed to get referenced strategy", err, updateErr)
+		}
+
+		return nil, err
 	}
 
-	if build.Spec.StrategyRef.Kind == nil || *build.Spec.StrategyRef.Kind == buildv1alpha1.NamespacedBuildStrategyKind {
-		buildStrategy, err := resources.RetrieveBuildStrategy(ctx, r.client, build)
+	switch *build.Spec.StrategyRef.Kind {
+	case buildv1alpha1.NamespacedBuildStrategyKind:
+		strategy, err = resources.RetrieveBuildStrategy(ctx, r.client, build)
 		if err != nil {
-			return nil, err
-		}
-		if buildStrategy != nil {
-			generatedTaskRun, err = resources.GenerateTaskRun(r.config, build, buildRun, serviceAccount.Name, buildStrategy)
-			if err != nil {
-				updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-				return nil, resources.HandleError("Failed to generate the taskrun with buildStrategy", err, updateErr)
+			if apierrors.IsNotFound(err) {
+				if updateErr := resources.UpdateConditionWithFalseStatus(ctx, r.client, buildRun, err.Error(), resources.BuildStrategyNotFound); updateErr != nil {
+					return nil, resources.HandleError("failed to get referenced strategy", err, updateErr)
+				}
 			}
 		}
-	} else if *build.Spec.StrategyRef.Kind == buildv1alpha1.ClusterBuildStrategyKind {
-		clusterBuildStrategy, err := resources.RetrieveClusterBuildStrategy(ctx, r.client, build)
+	case buildv1alpha1.ClusterBuildStrategyKind:
+		strategy, err = resources.RetrieveClusterBuildStrategy(ctx, r.client, build)
 		if err != nil {
-			return nil, err
-		}
-		if clusterBuildStrategy != nil {
-			generatedTaskRun, err = resources.GenerateTaskRun(r.config, build, buildRun, serviceAccount.Name, clusterBuildStrategy)
-			if err != nil {
-				updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-				return nil, resources.HandleError("Failed to generate the taskrun with clusterBuildStrategy", err, updateErr)
+			if apierrors.IsNotFound(err) {
+				if updateErr := resources.UpdateConditionWithFalseStatus(ctx, r.client, buildRun, err.Error(), resources.ClusterBuildStrategyNotFound); updateErr != nil {
+					return nil, resources.HandleError("failed to get referenced strategy", err, updateErr)
+				}
 			}
 		}
-	} else {
-		err := fmt.Errorf("unknown strategy %s", string(*build.Spec.StrategyRef.Kind))
-		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-		return nil, resources.HandleError(fmt.Sprintf("Unsupported BuildStrategy Kind: %v", build.Spec.StrategyRef.Kind), err, updateErr)
-	}
-
-	// Set OwnerReference for BuildRun and TaskRun
-	if err := r.setOwnerReferenceFunc(buildRun, generatedTaskRun, r.scheme); err != nil {
-		updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-		return nil, resources.HandleError("failed to set OwnerReference for BuildRun and TaskRun", err, updateErr)
+	default:
+		err = fmt.Errorf("unknown strategy %s", string(*build.Spec.StrategyRef.Kind))
```

Comment:
Is it possible to handle this with schema validation as well ? OpenAPI supports enums. We currently define BuildStrategyKind like this in buildstrategy.go:

```go
type BuildStrategyKind string
```

In typescript I would simply write:

```ts
type BuildStrategyKind = 'BuildStrategy' | 'ClusterBuildStrategy'
```

Not sure if something like this is possible in go and supported by the kube typing.

Reply:
Not sure. Im totally fine with the way it is, it actually allowed me to simplify this code into a switch clause.